pkg/controller/pxc/vs.go

```diff
@@ -4,18 +4,42 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"net/url"
 	"time"
 )
 
-func (vs VersionServiceClient) GetExactVersion(desiredVersion string, currentVersion string) (DepVersion, error) {
+func (vs VersionServiceClient) GetExactVersion(desiredVersion, currentVersion string, versionMeta currVersionMeta) (DepVersion, error) {
 	client := http.Client{
 		Timeout: 5 * time.Second,
 	}
-	url := fmt.Sprintf("%s/api/versions/v1/pxc/%s/%s?databaseVersion=%s", vs.URL, vs.OpVersion, desiredVersion, currentVersion)
-	req, err := http.NewRequest("GET", url, nil)
+
+	requestURL, err := url.Parse(fmt.Sprintf("%s/api/versions/v1/pxc/%s/%s", vs.URL, vs.OpVersion, desiredVersion))
+	if err != nil {
+		return DepVersion{}, err
+	}
+
+	q := requestURL.Query()
+	q.Add("databaseVersion", currentVersion)
+	q.Add("kubeVersion", versionMeta.KubeVersion)
+	q.Add("platform", versionMeta.Platform)
+	q.Add("customResourceUID", versionMeta.CRUID)
+
+	if versionMeta.PMMVersion != "" {
+		q.Add("pmmVersion", versionMeta.PMMVersion)
+	}
+
+	if versionMeta.BackupVersion != "" {
+		q.Add("backupVersion", versionMeta.BackupVersion)
+	}
+
+	requestURL.RawQuery = q.Encode()
+	req, err := http.NewRequest("GET", requestURL.String(), nil)
 	if err != nil {
 		return DepVersion{}, err
 	}
+
+	log.Info("FINAL URL", "URL", requestURL.String)
```

Comment:
can we log it on the Debug level? (not info)

Reply:
oops, sorry, forget to remove it)